.github/workflows/ci.yml

```diff
@@ -287,6 +288,44 @@ jobs:
       - run: make test.unit
 
 
+  test-e2e:
+    name: E2E tests
+    runs-on: ubuntu-latest
```

Comment:
@SleepySquash we should not test just some random build, but rather the Docker image we've built and we're going to ship. That's the point of E2E testing.

So here we should use `needs: ["docker"]`, and make sure the tests run against the downloaded image.

Reply:
@tyranron, actually, I'm not sure about it. `flutter drive` starts its own development server and runs E2E tests against it regardless of the Docker image passed in compose environment, so we're never testing the container itself (we're testing the server ran by `flutter drive`). Frontend is only used to route the compose services correctly (and `Dockerfile` only contains copying of `build/web` directory, configuration of `nginx` is the same). I see why we should use the built Docker image, but is it still worth using the `docker` artifact considering the above?

Reply:
@SleepySquash the point is in testing the same build going to ship, not concretely the Docker image. Is Flutter capable of it, or it works only with its buildserver?

Reply:
@tyranron, as far as I know, `flutter drive` is unable to test already built application. There's something called `Firebase Test Lab` listed in [the documentation of integration tests](https://docs.flutter.dev/testing/integration-tests), but it states it may be used to run tests only on iOS and Android platforms, not to say that it requires Firebase. It seems we might be able [to build the tests and run them afterwards](https://github.com/flutter/flutter/tree/main/packages/integration_test) (e.g. `flutter build integration_test/foo_test.dart`), but not the release app to run tests through them.

Reply:
@SleepySquash ok then, not much options here.